crates/bevy_render/src/texture/image.rs

```diff
@@ -106,7 +106,45 @@ pub struct Image {
     pub data: Vec<u8>,
     // TODO: this nesting makes accessing Image metadata verbose. Either flatten out descriptor or add accessors
     pub texture_descriptor: wgpu::TextureDescriptor<'static>,
-    pub sampler_descriptor: wgpu::SamplerDescriptor<'static>,
+    pub sampler_descriptor: ImageSampler,
+}
+
+#[derive(Debug, Clone)]
+pub enum ImageSampler {
+    Default,
+    Descriptor(wgpu::SamplerDescriptor<'static>),
+}
+impl Default for ImageSampler {
+    fn default() -> Self {
+        Self::Default
+    }
+}
+
+/// Resource used as the global default image sampler for [`Image`]s with their `sampler_descriptor`
+/// set to [`ImageSampler::Default`]. Otherwise, the specified sampler in
+/// [`ImageSampler::Descriptor`] will be used.
+#[derive(Debug, Clone)]
+pub struct DefaultImageSampler(pub wgpu::SamplerDescriptor<'static>);
+impl DefaultImageSampler {
+    pub fn linear() -> Self {
+        DefaultImageSampler(wgpu::SamplerDescriptor {
+            mag_filter: wgpu::FilterMode::Linear,
+            min_filter: wgpu::FilterMode::Linear,
+            ..Default::default()
+        })
+    }
+    pub fn nearest() -> Self {
+        DefaultImageSampler(wgpu::SamplerDescriptor {
+            mag_filter: wgpu::FilterMode::Nearest,
+            min_filter: wgpu::FilterMode::Nearest,
+            ..Default::default()
+        })
+    }
+}
+
+pub fn extract_image_sampler(mut commands: Commands, sampler: Res<DefaultImageSampler>) {
```

Comment:
Docs; this is public and somewhat mysterious.

Reply:
I'm looking to replace this with `ExtractResourcePlugin` but running into some errors.